presto-main/src/test/java/io/prestosql/tracer/TestTracer.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.tracer;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import io.airlift.json.JsonCodec;
+import io.prestosql.spi.eventlistener.TracerEvent;
+import io.prestosql.spi.tracer.DefaultTracer;
+import io.prestosql.spi.tracer.Tracer;
+import io.prestosql.spi.tracer.TracerEventType;
+import org.testng.annotations.Test;
+
+import java.net.URI;
+import java.util.Map;
+
+import static io.prestosql.spi.tracer.TracerEventType.PLAN_QUERY_END;
+import static io.prestosql.spi.tracer.TracerEventType.PLAN_QUERY_START;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertNull;
+import static org.testng.Assert.assertTrue;
+
+public class TestTracer
+{
+    private static final String QUERY_ID = "query";
+    private static final String STAGE_ID = "stage";
+    private static final String TASK_ID = "task";
+    private static final String NODE_ID = "node";
+    private static final URI TEST_URI = URI.create("http://test.com");
+    private static final TracerEventType QUERY_EVENT = PLAN_QUERY_START;
+    private static final TracerEventType TASK_EVENT = PLAN_QUERY_END;
+    private static final Map<String, Object> PAYLOAD = ImmutableMap.of("key", "value");
+
+    private final JsonCodec<Map<String, Object>> jsonCodec = JsonCodec.mapJsonCodec(String.class, Object.class);
```

Comment:
Consider making these variables in the `testTracer` method. There's no real reason for them to be class-level constants.